### PR TITLE
Octicons usage guidelines

### PR DESCRIPTION
**Pull request by ashygee**
This PR adds guidelines for Octicons to cover how to appropriately use them within GitHub's UI.

To do:
- [x] Add example images
- [x] Rewrite sizing section

### CODE REVIEW

**Comment by ashygee** [comment]
@yaili @edokoa @colebemis This is ready for review.

File: docs/content/guidelines/usage.mdx [hunk 1]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
...
-[TODO: Add image]
+Certain octicons are intended for a specific use case. Refer to the table below for icons with specific guidelines.
 
-### Spacing
+### Git specific
 
-When used in groups, octicons should have a minimum of 4px space between each icon.
+| Icon | Metaphor | Usage |
+| :--: | :--: | :--: |
+| <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path fill-rule="evenodd" d="M10.5 7.75a2.5 2.5 0 11-5 0 2.5 2.5 0 015 0zm1.43.75a4.002 4.002 0 01-7.86 0H.75a.75.75 0 110-1.5h3.32a4.001 4.001 0 017.86 0h3.32a.75.75 0 110 1.5h-3.32z"></path></svg> | **Commits** | The `commit` icon is represented as a circle on a line. This symbolizes a point (the circle) along the version history timeline in which changes were made. Use this icon in views to represent individual or groups of commits, including the timeline and the Commits tab in pull requests.
```
Comment (yaili) [inline docs/content/guidelines/usage.mdx:30]:
```suggestion
| <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path fill-rule="evenodd" d="M10.5 7.75a2.5 2.5 0 11-5 0 2.5 2.5 0 015 0zm1.43.75a4.002 4.002 0 01-7.86 0H.75a.75.75 0 110-1.5h3.32a4.001 4.001 0 017.86 0h3.32a.75.75 0 110 1.5h-3.32z"></path></svg> | **Commits** | The `commit` icon is represented as a circle on a line. This symbolizes a point (the circle) along the version history timeline when changes were made. Use this icon in views to represent individual or groups of commits, including the timeline and the **Commits** tab in pull requests.
```

File: docs/content/guidelines/usage.mdx [hunk 2]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
...
...8 12a1 1 0 100-2 1 1 0 000 2zm0-8a.75.75 0 01.75.75v3.5a.75.75 0 11-1.5 0v-3.5A.75.75 0 018 4zm4.78 4.28l3-3a.75.75 0 00-1.06-1.06l-2.47 2.47-.97-.97a.749.749 0 10-1.06 1.06l1.5 1.5a.75.75 0 001.06 0z"></path></svg> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16" class="octicon"><path fill-rule="evenodd" d="M8 1.5a6.491 6.491 0 00-5.285 2.715l1.358 1.358A.25.25 0 013.896 6H.25A.25.25 0 010 5.75V2.104a.25.25 0 01.427-.177l1.216 1.216a8 8 0 0114.315 4.03.748.748 0 01-.668.83.75.75 0 01-.824-.676A6.501 6.501 0 008 1.5zM.712 8.004a.75.75 0 01.822.67 6.501 6.501 0 0011.751 3.111l-1.358-1.358a.25.25 0 01.177-.427h3.646a.25.25 0 01.25.25v3.646a.25.25 0 01-.427.177l-1.216-1.216A8 8 0 01.042 8.827a.75.75 0 01.67-.823zM9 11a1 1 0 11-2 0 1 1 0 012 0zm-.25-6.25a.75.75 0 00-1.5 0v3.5a.75.75 0 001.5 0v-3.5z"></path></svg> | **Issues** | The `issue` icon is represented as a circle with an `!` mark. The base `issue` icon is extended through slight modifications that communicate different states of an issue: a checkmark for a closed issue, and by combining the sync icon with an `!` for reopening an issue. These three symbols should be used exclusively with issues.
```
Comment (yaili) [inline docs/content/guidelines/usage.mdx:31]: Adding code marks to the "sync" icon.

```suggestion
| <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path fill-rule="evenodd" d="M8 1.5a6.5 6.5 0 100 13 6.5 6.5 0 000-13zM0 8a8 8 0 1116 0A8 8 0 010 8zm9 3a1 1 0 11-2 0 1 1 0 012 0zm-.25-6.25a.75.75 0 00-1.5 0v3.5a.75.75 0 001.5 0v-3.5z"></path></svg> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path fill-rule="evenodd" d="M1.5 8a6.5 6.5 0 0110.65-5.003.75.75 0 00.959-1.153 8 8 0 102.592 8.33.75.75 0 10-1.444-.407A6.5 6.5 0 011.5 8zM8 12a1 1 0 100-2 1 1 0 000 2zm0-8a.75.75 0 01.75.75v3.5a.75.75 0 11-1.5 0v-3.5A.75.75 0 018 4zm4.78 4.28l3-3a.75.75 0 00-1.06-1.06l-2.47 2.47-.97-.97a.749.749 0 10-1.06 1.06l1.5 1.5a.75.75 0 001.06 0z"></path></svg> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16" class="octicon"><path fill-rule="evenodd" d="M8 1.5a6.491 6.491 0 00-5.285 2.715l1.358 1.358A.25.25 0 013.896 6H.25A.25.25 0 010 5.75V2.104a.25.25 0 01.427-.177l1.216 1.216a8 8 0 0114.315 4.03.748.748 0 01-.668.83.75.75 0 01-.824-.676A6.501 6.501 0 008 1.5zM.712 8.004a.75.75 0 01.822.67 6.501 6.501 0 0011.751 3.111l-1.358-1.358a.25.25 0 01.177-.427h3.646a.25.25 0 01.25.25v3.646a.25.25 0 01-.427.177l-1.216-1.216A8 8 0 01.042 8.827a.75.75 0 01.67-.823zM9 11a1 1 0 11-2 0 1 1 0 012 0zm-.25-6.25a.75.75 0 00-1.5 0v3.5a.75.75 0 001.5 0v-3.5z"></path></svg> | **Issues** | The `issue` icon is represented as a circle with an `!` mark. The base `issue` icon is extended through slight modifications that communicate different states of an issue: a checkmark for a closed issue, and by combining the `sync` icon with an `!` for reopening an issue. These three symbols should be used exclusively with issues.
```

File: docs/content/guidelines/usage.mdx [hunk 3]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
...
+## Special icons
 
-[TODO: Add image]
+Certain octicons are intended for a specific use case. Refer to the table below for icons with specific guidelines.
 
-### Spacing
+### Git specific
 
-When used in groups, octicons should have a minimum of 4px space between each icon.
+| Icon | Metaphor | Usage |
```
Comment (yaili) [inline docs/content/guidelines/usage.mdx:28]: Is "metaphor" the best heading for these tables? Can we simplify or is that a common way to list icon meanings? 🤔

File: docs/content/guidelines/usage.mdx [hunk 4]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
...
 
-[TODO: Add image]
+### GitHub brand specific
 
-## Specific icon usage
+| Icon | Metaphor | Usage |
+| :--: | :--: | :--: |
+| <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path fill-rule="evenodd" d="M1.326 1.973a1.2 1.2 0 011.49-.832c.387.112.977.307 1.575.602.586.291 1.243.71 1.7 1.296.022.027.042.056.061.084A13.22 13.22 0 018 3c.67 0 1.289.037 1.861.108l.051-.07c.457-.586 1.114-1.004 1.7-1.295a9.654 9.654 0 011.576-.602 1.2 1.2 0 011.49.832c.14.493.356 1.347.479 2.29.079.604.123 1.28.07 1.936.541.977.773 2.11.773 3.301C16 13 14.5 15 8 15s-8-2-8-5.5c0-1.034.238-2.128.795-3.117-.08-.712-.034-1.46.052-2.12.122-.943.34-1.797.479-2.29zM8 13.065c6 0 6.5-2 6-4.27C13.363 5.905 11.25 5 8 5s-5.363.904-6 3.796c-.5 2.27 0 4.27 6 4.27z"></path><path d="M4 8a1 1 0 012 0v1a1 1 0 01-2 0V8zm2.078 2.492c-.083-.264.146-.492.422-.492h3c.276 0 .505.228.422.492C9.67 11.304 8.834 12 8 12c-.834 0-1.669-.696-1.922-1.508zM10 8a1 1 0 112 0v1a1 1 0 11-2 0V8z"></path></svg> | **Octoface** | Represents the user. Conveys connection between GitHub and the user. Similar to the octocat generator.
```
Comment (yaili) [inline docs/content/guidelines/usage.mdx:39]: What's the "octocat generator"?

File: docs/content/guidelines/usage.mdx [hunk 5]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
...
+| <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path fill-rule="evenodd" d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z"></path></svg> | **GH mark** | Brand logo use only. Follow [GitHub's brand guidelines](https://brand.github.com/) when using this octicon.
 
-### Octicons custom to `git`
+## Icon styling
 
-The following should only be used for their specific meanings:
+By default, all Octicons use an outline style. Some icons have a filled version which should be use only under the following conditions:
```
Comment (yaili) [inline docs/content/guidelines/usage.mdx:44]:
```suggestion
By default, all octicons use an outline style. Some icons have a filled version which should be used only under the following conditions:
```

File: docs/content/guidelines/usage.mdx [hunk 6]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
...
-- Issues: The issue icon is represented as a circle with an `!` mark. The base issue icon is extended through slight modifications to communicate different states of an issue: a checkmark for a closed issue and combining the sync icon with an `!` for reopening an issue. These three symbols should be used exclusively with issues.
-- Pull requests: The pull request icon is represented as a commit dot attached with an arrow being inserted into the history timeline (two dots connected by a line).
-- Repos: A repository in GitHub is represented as a closed book and is explicitly named as `repo`. Octicons does contain a `book` icon which displays an open book symbol that is used for wikis, READMEs, and other use cases where the user can see a collection of information.
+<Do>
+    <img width="456" alt="octicons-style-pairing" src="https://user-images.githubusercontent.com/10384315/103719162-490c9600-4f7d-11eb-8d9c-1d867ee51de9.png" />
+    <Caption>To aid in contrast when in relation to another icon</Caption>
```
Comment (yaili) [inline docs/content/guidelines/usage.mdx:48]:
```suggestion
    <Caption>Use a filled icon to aid in contrast when in relation to another icon</Caption>
```

File: docs/content/guidelines/usage.mdx [hunk 7]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
...
-- To show state [TODO: Screenshot of star and sponsor social buttons]
-
-### Status
-
-To show the status or completion of an action (e.g. an action running), the filled check and filled x icons can be used.
-
-[TODO: Show image of Actions]
+<Do>
+    <img width="456" alt="octicon-mobile-state" src="https://user-images.githubusercontent.com/10384315/103719160-4873ff80-4f7d-11eb-8253-634e1692d96b.png" />
+    <Caption>To indicate navigation state in GitHub's native mobile apps</Caption>
```
Comment (yaili) [inline docs/content/guidelines/usage.mdx:53]:
```suggestion
    <Caption>Use a filled icon to indicate navigation state in GitHub's native mobile apps</Caption>
```

File: docs/content/guidelines/usage.mdx [hunk 8]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
...
-- Issues: The issue icon is represented as a circle with an `!` mark. The base issue icon is extended through slight modifications to communicate different states of an issue: a checkmark for a closed issue and combining the sync icon with an `!` for reopening an issue. These three symbols should be used exclusively with issues.
-- Pull requests: The pull request icon is represented as a commit dot attached with an arrow being inserted into the history timeline (two dots connected by a line).
-- Repos: A repository in GitHub is represented as a closed book and is explicitly named as `repo`. Octicons does contain a `book` icon which displays an open book symbol that is used for wikis, READMEs, and other use cases where the user can see a collection of information.
+<Do>
+    <img width="456" alt="octicons-style-pairing" src="https://user-images.githubusercontent.com/10384315/103719162-490c9600-4f7d-11eb-8d9c-1d867ee51de9.png" />
+    <Caption>To aid in contrast when in relation to another icon</Caption>
+</Do>
```
Comment (yaili) [inline docs/content/guidelines/usage.mdx:50]: I noticed there's no margin between these 2 do components, at least in [this version](https://octicons-e76mon3x5.vercel.app/octicons/guidelines/usage). Is that something we need to add manually?

File: docs/content/guidelines/usage.mdx [hunk 9]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
...
-The state of an icon can be displayed using outline/fill icon pairs or by using an icon's slash variant. Read below to understand the difference.
+The state of an action can be displayed either by using a stroke and fill icon pairing or using a slash version of an icon if available.
 
 ### Outline and fill pairings
 
 Use the outline icon for the "off"/"empty" state and the fill icons for "on"/"filled" states.
 
-[Example from mobile nav bar]
-[Example from sponsors button]
+<Note>The exception to this rule is when conveying a "pass"/"fail" pairing which is done using the filled version of the `check-circle` and `x-circle`</Note>
```
Comment (yaili) [inline docs/content/guidelines/usage.mdx:64]: The code blocks aren't rendering correctly inside the note component. Is this a bug we should fix now?

File: docs/content/guidelines/usage.mdx [hunk 10]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
...
+| Pass/Fail | <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path fill-rule="evenodd" d="M8 16A8 8 0 108 0a8 8 0 000 16zm3.78-9.72a.75.75 0 00-1.06-1.06L6.75 9.19 5.28 7.72a.75.75 0 00-1.06 1.06l2 2a.75.75 0 001.06 0l4.5-4.5z"></path></svg> / <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path fill-rule="evenodd" d="M2.343 13.657A8 8 0 1113.657 2.343 8 8 0 012.343 13.657zM6.03 4.97a.75.75 0 00-1.06 1.06L6.94 8 4.97 9.97a.75.75 0 101.06 1.06L8 9.06l1.97 1.97a.75.75 0 101.06-1.06L9.06 8l1.97-1.97a.75.75 0 10-1.06-1.06L8 6.94 6.03 4.97z"></path></svg> |
+
+<Do>
+    <img width="456" alt="octicons-state" src="https://user-images.githubusercontent.com/10384315/103719161-490c9600-4f7d-11eb-9e44-8bafc3fd64b5.png" />
+    <Caption>To show state of an action</Caption>
```
Comment (yaili) [inline docs/content/guidelines/usage.mdx:74]:
```suggestion
    <Caption>Use outline and fill pairings to show the state of an action</Caption>
```

File: docs/content/guidelines/usage.mdx [hunk 11]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
...
+</Do>
+
+<Do>
+    <img width="456" alt="octicon-actions" src="https://user-images.githubusercontent.com/10384315/103720006-3d21d380-4f7f-11eb-89b8-3b572f80e284.png"/>
+    <Caption>To indicate the status or completion of an action (for example, an action running).</Caption>
```
Comment (yaili) [inline docs/content/guidelines/usage.mdx:79]:
```suggestion
    <Caption>Use outline and fill pairings to indicate the status or completion of an action (for example, an action running)</Caption>
```

**Review by yaili — CHANGES_REQUESTED** [review]
It's looking good. Just some small suggestions, and a few questions it would be good to clarify. ❤️

File: docs/content/guidelines/usage.mdx [hunk 12]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
 import {Caption, Do, Dont} from '@primer/gatsby-theme-doctocat'
 import {Grid, Flex, Box} from '@primer/components'
 
-# Octicons usage guidelines
+## General guidelines
```
[the remark this answers is not in the document]
Comment (ashygee) [reply docs/content/guidelines/usage.mdx:8]: Yeah that makes sense to me. I think I was trying to categorize that bullet list but having it as introductory information feels right. I'll update.

File: docs/content/guidelines/usage.mdx (hunk 3 above)
Comment (ashygee) [reply docs/content/guidelines/usage.mdx:28]: Metaphor is how we referred to it during the process and I saw that Adobe Spectrum mentions metaphors too. Maybe it feels better to say something like, `Icon | Icon name | Usage | Guidelines`

File: docs/content/guidelines/usage.mdx (hunk 4 above)
Comment (ashygee) [reply docs/content/guidelines/usage.mdx:39]: https://myoctocat.com/. I will crosslink.

File: docs/content/guidelines/usage.mdx (hunk 9 above)
Comment (ashygee) [reply docs/content/guidelines/usage.mdx:64]: @colebemis can you look into this?

File: docs/content/guidelines/usage.mdx [hunk 13]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
...
+## Sizing
 
-These are general guidelines to follow when using octicons across various platforms:
+Octicons are available in two sizes: 16px and 24px. They should only be used at their dedicated sizes to maintain consistency in stroke width and legibility.
 
-1. Octicons should be used to _emphasize_ meaning, not replace.
-2. Adding an octicon should be intentional.
-3. Not everything needs an icon. Consider a text label before adding an icon.
-4. Octicons should be kept at their 1:1 aspect ratio
+<img width="960" alt="octicons-scaling" src="https://user-images.githubusercontent.com/10384315/103715870-87eb1d80-4f76-11eb-8b54-c25ba09ecb5f.png" />
```
Comment (colebemis) [inline docs/content/guidelines/usage.mdx:19]: It isn't clear to me what this image is trying to communicate. If we're trying to say "avoid scaling 16px and 24px icons," then a do/don't example might be more clear.

File: docs/content/guidelines/usage.mdx [hunk 14]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
...
-
-### Status
-
-To show the status or completion of an action (e.g. an action running), the filled check and filled x icons can be used.
-
-[TODO: Show image of Actions]
+<Do>
+    <img width="456" alt="octicon-mobile-state" src="https://user-images.githubusercontent.com/10384315/103719160-4873ff80-4f7d-11eb-8253-634e1692d96b.png" />
+    <Caption>Use a filled icon to indicate navigation state in GitHub's native mobile apps</Caption>
+</Do>
```
Comment (colebemis) [inline docs/content/guidelines/usage.mdx:54]: I think a "don't" example might be helpful here to show when not to use a filled icon.

File: docs/content/guidelines/usage.mdx [hunk 15]
```diff
@@ -5,74 +5,93 @@ title: Usage guidelines
...
 import {Grid, Flex, Box} from '@primer/components'
 
-# Octicons usage guidelines
+## General guidelines
 
-The following guidelines are to be used as a reference for how to use octicons.
+1. Octicons are used to _emphasize_ meaning, not as a replacement.
+2. Adding an octicon is intentional.
+3. An icon is need. Not everything needs an icon: consider a text label before adding an icon.
+4. Octicons are kept at their 1:1 aspect ratio.
```
Comment (colebemis) [inline docs/content/guidelines/usage.mdx:13]: Could we use a more active voice here? For example:

```diff
- Octicons are kept at their 1:1 aspect ratio.
+ Keep octicons at their aspect ratio.
```

File: docs/content/guidelines/usage.mdx (hunk 9 above)
Comment (colebemis) [reply docs/content/guidelines/usage.mdx:64]: This should make it work:

```suggestion
<Note>

The exception to this rule is when conveying a "pass"/"fail" pairing which is done using the filled version of the `check-circle` and `x-circle`

</Note>
```

**Review by colebemis — CHANGES_REQUESTED** [review]

File: docs/content/guidelines/usage.mdx (hunk 15 above)
Comment (ashygee) [reply docs/content/guidelines/usage.mdx:13]: Yes! Thank you for rewording that. I was struggling with it.

File: docs/content/guidelines/usage.mdx (hunk 13 above)
Comment (ashygee) [reply docs/content/guidelines/usage.mdx:19]: I wanted to communicate the change in stroke weight and legibility. I can change this do a do/don't example though.

File: docs/content/guidelines/usage.mdx (hunk 14 above)
Comment (ashygee) [reply docs/content/guidelines/usage.mdx:70]: Looking at the set we have I can't think of a realistic example to show here. If you have an idea please feel free to share!

File: docs/content/guidelines/usage.mdx (hunk 3 above)
Comment (yaili) [reply docs/content/guidelines/usage.mdx:44]: I don't have strong feelings about it, up to you.

File: docs/content/guidelines/usage.mdx (hunk 6 above)
Comment (yaili) [reply docs/content/guidelines/usage.mdx:65]: @ashygee Did you see this one?

**Review by yaili — APPROVED** [review]
Looks good, just noticed one of the suggestions wasn't resolved, did you see it? https://github.com/primer/octicons/pull/517/files#r552649466

File: docs/content/guidelines/usage.mdx (hunk 6 above)
Comment (ashygee) [reply docs/content/guidelines/usage.mdx:65]: Oh I must have missed this one

**Review by colebemis — APPROVED** [review]
Nice work, @ashygee!

**Pull request merged**